packages/data/lib/database/database_repository_impl.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:data/configuration/configuration_database.dart';
+import 'package:data/database/database_model/cast_model.dart';
+import 'package:data/database/database_model/movie_list_model.dart';
+import 'package:data/mappers/cast_mapper.dart';
+import 'package:data/mappers/movie_list_mapper.dart';
+import 'package:domain/entity/movie_list_response/movie_list_response.dart';
+import 'package:domain/model/response_model_people.dart';
+import 'package:domain/repository/database_repository.dart';
+import 'package:domain/use_case/request_movie_list_use_case.dart';
+import 'package:flutter/foundation.dart';
+import 'package:path/path.dart';
+import 'package:sqflite/sqflite.dart';
+import 'package:sqflite/sqlite_api.dart';
+
+class DatabaseRepositoryImpl implements DatabaseRepository {
+  static final DatabaseRepositoryImpl instance = DatabaseRepositoryImpl();
+  final MovieListMapper _movieListMapper = MovieListMapper();
+  final CastMapper _castMapper = CastMapper();
+  static Database? _database;
```

Comment:
Of course, it’s impossible to make the same service as dio, simply because there is no way to use insert/delete/query one and the same, but I’ll try something